app/models/data/high_fail_questions.rb

```diff
@@ -0,0 +1,50 @@
+module Data
+  class HighFailQuestions
+    include ToCsv
+
+    class << self
+      # @return [Array<String>]
+      def column_names
+        ['Module', 'Question', 'Failure Rate Percentage']
+      end
+
+      # @return [Array<Hash{Symbol => Mixed}>]
+      def dashboard
+        data = []
+        high_fail_questions.each do |(module_name, question_name), fail_rate|
+          row = {
+            module_name: module_name,
+            question_name: question_name,
+            fail_rate_percentage: fail_rate,
+          }
+          data << row
+        end
+        data
+      end
+
+  private
+
+      # @return [Hash{Symbol => Float, Array<Array<String, String>> => Float}]
+      def high_fail_questions
+        question_attempts = UserAnswer.summative.group(:module, :name).count
```

Comment:
I'd like to try refactoring this large method by moving the first variables into  private methods like we do in other models.

Reply:
Like this? [0c368b6](https://github.com/DFE-Digital/early-years-foundation-recovery/pull/648/commits/0c368b6eb789a1cf6247a6bd7d09d4d90a2cfd9b)